server/server.go

```diff
@@ -169,137 +297,23 @@ func (s *server) ErrorEncoder() kithttp.ErrorEncoder {
 
 		// Write the actual response body.
 		json.NewEncoder(w).Encode(map[string]interface{}{
-			"error": errMessage,
-			"from":  s.ServiceName(),
-		})
-	}
-}
-
-// NewRouter returns a HTTP handler for the server. Here we register all
-// endpoints listed in the endpoint collection.
-func (s *server) NewRouter() *mux.Router {
-	router := mux.NewRouter()
-
-	// Define our custom not found handler. Here we take care about logging,
-	// metrics and a proper response.
-	router.NotFoundHandler = http.Handler(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		// Log the error and its message. This is really useful for debugging.
-		errDomain := errorDomain(nil)
-		errMessage := fmt.Sprintf("not found: %s %s", r.Method, r.URL.Path)
-		errTrace := ""
-		s.logger.Log("error", map[string]string{"domain": errDomain, "message": errMessage, "trace": errTrace})
-
-		// This defered callback will be executed at the very end of the request.
-		defer func(t time.Time) {
-			endpointCode := strconv.Itoa(http.StatusNotFound)
-			endpointMethod := strings.ToLower(r.Method)
-			endpointName := "notfound"
-
-			endpointTotal.WithLabelValues(endpointCode, endpointMethod, endpointName).Inc()
-			endpointTime.WithLabelValues(endpointCode, endpointMethod, endpointName).Set(float64(time.Since(t) / time.Millisecond))
-
-			errorTotal.WithLabelValues(errDomain).Inc()
-		}(time.Now())
-
-		// Write the actual response body.
-		w.WriteHeader(http.StatusNotFound)
-		w.Header().Set("Content-Type", "application/json; charset=utf-8")
-		json.NewEncoder(w).Encode(map[string]interface{}{
-			"error": errMessage,
+			"code":  responseError.Code(),
```

Comment:
I want to enable microservices to provide a proper response including machine readable codes. This standard applies then across all of our services.